### PR TITLE
Timed traffic lights step - confirm before delete

### LINKED ISSUE #283: Dialog to confirm delete of timed traffic light step
From user Domaldel in Discord chat:

> I'd love for a pop up window asking for confirmation before deleting steps in a timed traffic light that involves multiple junctions.
> Trying to coordinate 18 steps across 13 junctions involving trains (going through the city), cars and pedestrians is proving a bit of a pain when all of a sudden delete a step involving all 13 junctions...
> The third time that has happened in that one city alone by the way...
> In a single day...

> Deleting a step for a single junction is just fine.
> You can just recreate it.
> But when you have multiple that all involves several steps the numbers of junction states that you have to juggle just becomes too high to easily recreate.
> And I don't remember the details, but basically there's a situation where you click on one button and the delete button ends up in the same exact spot a moment afterwards, so you end up clicking on it without intending to.
> That's both issues that should be fixed asap in my view.
> What I assume is a relatively easy fix would be to simply add a extra step the process of deleting a steps for timed traffic lights.
> When there's more then one junction involved.
> A simple "Are you sure that you want to delete step (number) for this and all connected junctions?" "Yes" or "No".
> Move the "delete" code over to the "yes" button instead of the "delete" button itself, possibly adding something to transfer variables over, etc.

### PR DESCRIPTION
**Pull request by krzychu124**
Fixes #283 

_Key features:_

- additional confirm button when user wants to remove timed traffic lights step
- right click on _Confirm_ to cancel action + auto cancel when clicked on any other button
- hold `Ctrl` while click on _Delete_ button to force removal without confirmation
- added translation keys

_TODO:_
- [x] change placement of delete button to minimize accidental click

### CODE REVIEW

**Comment by aubergine10** [comment]
LGTM, but I think we should seek additional community feedback before merging.

The problem as described is that after saving a step, the delete button appears where the save button was, and it's too easy to accidentally click it = that step you just added goes bye bye.

_(Note: The root issue is that there's no way to undo deleting a step, however implementing an undo feature is probably best left until we get round to completely rewriting the ancient traffic lights UI code at a later date.)_

Current functionality proposed in this PR:

SPOILER_screen150

@DaEgi01 commented that it makes UI cumbersome:

> never hand issues with step deletions ... now it'll be one more click ://

Domaldel mentioned maybe changing the position of the button:

> The issue was that the "save step" button and the "delete" button was right behind each other.  
> ...  
> Did you guys end up moving the confirmation button a bit so it's not at the same exact spot as the delete button?

File: TLM/TLM/Resources/lang_zh.txt [hunk 1]
```diff
@@ -228,4 +228,6 @@ Node_is_level_crossing 此节点级别交汇，不可禁用红绿灯。
 Experimental_features Experimental features
 Turn_on_red Turn on red
 Vehicles_may_turn_on_red 车辆红灯可转弯
-Also_apply_to_left/right_turns_between_one-way_streets Also apply to left & right turns between one-way streets
\ No newline at end of file
+Also_apply_to_left/right_turns_between_one-way_streets Also apply to left & right turns between one-way streets
+Timed_traffic_lights_manager_confirm Confirm!
```
Comment (Emphasia) [inline TLM/TLM/Resources/lang_zh.txt]:
```suggestion
Timed_traffic_lights_manager_confirm 请确认！
```

File: TLM/TLM/Resources/lang_zh.txt [hunk 2]
```diff
@@ -228,4 +228,6 @@ Node_is_level_crossing 此节点级别交汇，不可禁用红绿灯。
...
+Timed_traffic_lights_manager_right_click_to_cancel Right click to cancel
```
Comment (Emphasia) [inline TLM/TLM/Resources/lang_zh.txt]:
```suggestion
Timed_traffic_lights_manager_right_click_to_cancel 右键取消
```

**Comment by aubergine10** [comment]
With the button move, I'm not sure we need the confirmation step any more. Any accidental click will hit the View button so there's little to no risk of accidental deletion.

**Comment by krzychu124** [comment]
Changed placement of delete button should prevent accidental click so I removed delete confirmation step and removed unused translation keys

**Comment by aubergine10** [comment]
testing

**Comment by aubergine10** [comment]
The delete button no longer has red text - can that be re-added?

Otherwise looks good.

**Pull request merged**